x-pack/plugins/transform/common/utils/transform_api_key.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { GrantAPIKeyResult } from '@kbn/security-plugin/server';
+import type { SecurityCreateApiKeyResponse } from '@elastic/elasticsearch/lib/api/types';
+
+export interface TransformAPIKey extends GrantAPIKeyResult {
+  /**
+   * Generated encoded API key used for headers
+   */
+  encoded: string;
+}
+
+export interface SecondaryAuthorizationHeader {
+  headers?: { 'es-secondary-authorization': string | string[] };
+}
+
+export function isTransformApiKey(arg: any): arg is TransformAPIKey {
+  return (
+    arg &&
```

Comment:
You could do `isPopulated(arg, ['api_key', 'encoded']) && typeof arg.encoded === string` here.

Reply:
Updated here [`3f64b1f` (#154736)](https://github.com/elastic/kibana/pull/154736/commits/3f64b1f53667da00f811cc2579eb31decf81fafb)